### DISCUSSION TITLE
It's a good idea to put the react directive "use client" at the top of the mantine packages index .js files?

### DISCUSSION THREAD

**Discussion by MiiZZo** [mantinedev/mantine#4615] [Feedback]
The problem is that it would be better for me to re-export mantine components like this:
```ts
// shared/button.js
'use client'
export { Button } from '@mantine/core'
```
Instead of not being able to use any mantine package code in the server components. For example, I can't use the simple `createTheme` function from `@mantine/core` just because it's imported from index.a js file containing the "use client" directive. I know that the react ecosystem is not ready to work with server components, but it would be better to let developers integrate mantine with server components themselves.

**Comment by rtivital** [comment]
Okay, so here are some statistics:
- 80% of people here and on Discord/GitHub tell me that I should add `'use client';`. (this includes people that do not know what server components are and just mess around with app router)
- 20% of people tell me that I should not include `'use client';` and they will handle it on their side.

I (as a lead maintainer) do not care about the `'use client';` at all – I do not have any plans of using app router in my production applications any time soon. So my actions should go towards decreasing the number of GitHub issues and help requests. Based on my observations of people who are using v7, the only issue is with the `createTheme` function – no other complaints. That is good enough for me for now. You are welcome to propose a better solution if you know any.

**Reply by traviswimer** [reply]
Although I understand that it's not a high priority, is there any downside to adding this directive?

Since this is a React feature (not just NextJS), I would think that this will be a growing complaint as server components are utilized by more frameworks.

_____
Update:

_In case anyone else sees this and is confused like I was, V7 of Mantine is in a [separate repo](https://github.com/mantinedev/mantine-v7), and `'use client'` is automatically added to all the components in the [Rollup config](https://github.com/mantinedev/mantine-v7/blob/a1166e57ad1f4ca8d722348025ad63aea8d7611e/configuration/rollup/create-package-config.ts#L56)._

**Reply by rtivital** [reply]
`'use client';` is there in index file starting from v7 – all components are considered client.
